daemon/config.go

```diff
@@ -501,12 +501,34 @@ func ValidateConfiguration(config *Config) error {
 	if config.IsValueSet("max-concurrent-downloads") && config.MaxConcurrentDownloads != nil && *config.MaxConcurrentDownloads < 0 {
 		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
 	}
+	if config.MaxConcurrentDownloads ==nil {
+		maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		config.MaxConcurrentDownloads = &maxConcurrentDownloads
+	}
+	if *config.MaxConcurrentDownloads < 1 {
+		//maxConcurrentDownloads :=defaultMaxConcurrentDownloads
+		//config.MaxConcurrentDownloads = &maxConcurrentDownloads
+		return fmt.Errorf("invalid max concurrent downloads: %d", *config.MaxConcurrentDownloads)
```

Comment:
Isn't this doing what https://github.com/docker/docker/pull/28279/files#diff-a348195a1b645d5b477946b1efae728bR502 already does?